### PR TITLE
[dagster-pandas] create a metadata schema table function

### LINKED ISSUE #6957: add a utility for creating table schema metadata for a Pandas dataframe

### PR DESCRIPTION
**Pull request by AndyBys**
### Summary & Motivation

Closes #6957 

### How I Tested These Changes

[Test](https://github.com/dagster-io/dagster/blob/de568cb387e61d3f513579fe9ee13670e86c2329/python_modules/libraries/dagster-pandas/dagster_pandas_tests/test_pandas_dataframe_metadata.py)

### CODE REVIEW

File: python_modules/libraries/dagster-pandas/dagster_pandas/data_frame.py [hunk 1]
```diff
@@ -120,6 +125,31 @@ def create_dagster_pandas_dataframe_description(description, columns):
...
+    This function takes a pandas DataFrame and returns its metadata as a Dagster TableSchema.
+
+    Args:
+        pandas_df (pandas.DataFrame): A pandas DataFrame for which to create metadata.
+
+    Returns:
+        Union[TypeCheck, MetadataValue]: If the input is not a pandas DataFrame, it returns a TypeCheck object
+        with success=False. Otherwise, it returns a MetadataValue object with the TableSchema for the DataFrame.
+    """
+    if not isinstance(pandas_df, pd.DataFrame):
```
Comment (smackesey) [inline python_modules/libraries/dagster-pandas/dagster_pandas/data_frame.py:141]: Ah, I think you've misunderstood the purpose of Dagster's `TypeCheck`-- this is a framework-specific entity used to encapsulate typechecks for op inputs/outputs, not a generic entity used to represent a typecheck for a python function.

Based on my understanding of the purpose of this function, pretty sure the `TypeCheck` case should just be removed. Use `check.inst(pandas_df, pd.DataFrame)` to verify the input is a dataframe and you can drop the conditional.

File: python_modules/libraries/dagster-pandas/dagster_pandas_tests/test_pandas_dataframe_metadata.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+def test_create_dagster_pandas_dataframe_metadata():
+    # Test with a valid DataFrame
+    df = DataFrame({"col1": [1, 2], "col2": ["a", "b"]})
+    expected_table_schema = TableSchema(
+        columns=[TableColumn(name="col1", type="int64"), TableColumn(name="col2", type="object")]
+    )
+    assert create_dagster_pandas_dataframe_metadata(df) == MetadataValue.table_schema(expected_table_schema)
+
+    # Test with an invalid input
+    invalid_input = "not a DataFrame"
```
Comment (smackesey) [inline python_modules/libraries/dagster-pandas/dagster_pandas_tests/test_pandas_dataframe_metadata.py:15]: You can drop the invalid input test case per above comment.

**Review by smackesey — CHANGES_REQUESTED** [review]

File: python_modules/libraries/dagster-pandas/dagster_pandas/data_frame.py (hunk 1 above)
Comment (AndyBys) [reply python_modules/libraries/dagster-pandas/dagster_pandas/data_frame.py]: Thanks for details on this topic. Fixed

File: python_modules/libraries/dagster-pandas/dagster_pandas_tests/test_pandas_dataframe_metadata.py (hunk 2 above)
Comment (AndyBys) [reply python_modules/libraries/dagster-pandas/dagster_pandas_tests/test_pandas_dataframe_metadata.py]: Solved

File: python_modules/libraries/dagster-pandas/dagster_pandas/data_frame.py [hunk 3]
```diff
@@ -120,6 +125,30 @@ def create_dagster_pandas_dataframe_description(description, columns):
...
+    pandas_df: pd.DataFrame,
+) -> Union[TypeCheck, MetadataValue]:
+    """
+    This function takes a pandas DataFrame and returns its metadata as a Dagster TableSchema.
+
+    Args:
+        pandas_df (pandas.DataFrame): A pandas DataFrame for which to create metadata.
+
+    Returns:
+        Union[TypeCheck, MetadataValue]: If the input is not a pandas DataFrame, it returns a TypeCheck object
```
Comment (sryza) [inline python_modules/libraries/dagster-pandas/dagster_pandas/data_frame.py:138]: I see that you updated the code to remove the `TypeCheck` case, but it needs to be removed from the docstring as well.

**Review by sryza — COMMENTED** [review]
Thanks for the update on this @AndyBys! I added a couple more comments. After those are addressed, this should be good to go.

**Comment by AndyBys** [comment]
Hey @sryza, thanks for feedback. I have made changes to address issues. Should be fine now.

**Comment by AndyBys** [comment]
> Do you mind running `make ruff` in the Dagster root directory? That will sort the import statements so that the code passes Dagster's automated checks. After that, this should be good to go.

Done.

**Review by sryza — APPROVED** [review]
This looks great. Thanks @AndyBys !

**Pull request merged**